addons/l10n_be_hr_payroll_fleet/models/fleet.py

```diff
@@ -45,13 +45,23 @@ def _compute_total_cost(self):
                 elif contract.cost_frequency == "yearly":
                     car.total_cost += contract.cost_generated / 12.0
 
-    def _get_co2_fee(self, co2):
-        return max((((co2 * 9.0) - 600.0) * 1.2488) / 12.0, 0.0)
-
-    @api.depends('co2')
+    def _get_co2_fee(self, co2, fuel_type):
+        fuel_coefficient = 600
+        if fuel_type == 'diesel':
```

Comment:
Maybe set fuel_coefficient as a dictionary and make ``fuel_coefficient.get(fuel_type)``